Sources/protoc-gen-swift/SwiftGeneratorPlugin.swift

```diff
@@ -35,14 +35,29 @@ struct SwiftGeneratorPlugin: CodeGenerator {
         var errorString: String? = nil
         for fileDescriptor in files {
             let fileGenerator = FileGenerator(fileDescriptor: fileDescriptor, generatorOptions: options)
-            var printer = CodePrinter(addNewlines: true)
-            fileGenerator.generateOutputFile(printer: &printer, errorString: &errorString)
+            var filePrinter = FilePrinter(
+                main: CodePrinter(addNewlines: true),
+                hashable: CodePrinter(addNewlines: true),
+                nameProviding: CodePrinter(addNewlines: true)
+            )
+            fileGenerator.generateOutputFile(filePrinter: &filePrinter, errorString: &errorString)
             if let errorString = errorString {
                 // If generating multiple files, scope the message with the file that triggered it.
                 let fullError = files.count > 1 ? "\(fileDescriptor.name): \(errorString)" : errorString
                 throw GenerationError.message(message: fullError)
             }
-            try generatorOutputs.add(fileName: fileGenerator.outputFilename, contents: printer.content)
+            try generatorOutputs.add(
+                fileName: fileGenerator.outputFilename(".pb.swift"),
+                contents: filePrinter.main.content
+            )
+            try generatorOutputs.add(
+                fileName: fileGenerator.outputFilename(".pb.hashable.swift"),
```

Comment:
In the SwiftProtobuf sources, we have the convention that a file declaring an extension of some type gets named "Type+Protocol.swift"

I wonder if some variant of that convention would be more appropriate here?  I think that would end up giving us ".pb+Hashable.swift" which is admittedly a little weird looking.   But at a minimum, `Hashable` and `NameProviding` in these filenames should be correctly capitalized.